src/server/server.rs

```diff
@@ -34,15 +41,27 @@ pin_project! {
     }
 }
 
+/// A listening HTTP server that accepts connections in both HTTP1 and HTTP2 by default.
+///
+/// Needs at least one of the `http1` and `http2` features to be activated to actually be useful.
+#[cfg(not(any(feature = "http1", feature = "http2")))]
+pub struct Server<I, S, E = Exec> {
```

Comment:
Why should we expose a type that can't do anything if the features aren't enabled?

Reply:
This allows axum to unconditionally re-export the type. It could also conditionally re-export I suppose¹, but it seemed logical that hyper would always define `hyper::Server` when the equally-named feature is enabled 🙂

¹ which would mean having to enable at least one of its "feature re-exports" for http1 / http2 in order to use the re-export, as opposed to enabling one or both directly; certainly not such a bad thing in the grand scheme of things but potentially a small papercut

Reply:
Oh I get it now. I should have better read the description and linked issue, my bad!